src/tracer.cc

```diff
@@ -33,34 +33,50 @@
 #ifdef TRITON_ENABLE_GPU
 #include <cuda_runtime_api.h>
 #endif  // TRITON_ENABLE_GPU
+#ifndef _WIN32
+namespace otlp = opentelemetry::exporter::otlp;
+namespace otel_trace_sdk = opentelemetry::sdk::trace;
+namespace otel_trace_api = opentelemetry::trace;
+namespace otel_common = opentelemetry::common;
+#endif
 
 namespace triton { namespace server {
 
 TRITONSERVER_Error*
 TraceManager::Create(
     TraceManager** manager, const TRITONSERVER_InferenceTraceLevel level,
     const uint32_t rate, const int32_t count, const uint32_t log_frequency,
-    const std::string& filepath)
+    const std::string& filepath, const InferenceTraceMode mode,
+    const triton::server::TraceConfigMap& config_map)
 {
   // Always create TraceManager regardless of the global setting as they
   // can be updated at runtime even if tracing is not enable at start.
   // No trace should be sampled if the setting is not valid.
-  *manager = new TraceManager(level, rate, count, log_frequency, filepath);
+  *manager = new TraceManager(
+      level, rate, count, log_frequency, filepath, mode, config_map);
+
   return nullptr;  // success
 }
 
 TraceManager::TraceManager(
     const TRITONSERVER_InferenceTraceLevel level, const uint32_t rate,
     const int32_t count, const uint32_t log_frequency,
-    const std::string& filepath)
+    const std::string& filepath, const InferenceTraceMode mode,
+    const TraceConfigMap& config_map)
 {
   std::shared_ptr<TraceFile> file(new TraceFile(filepath));
   global_default_.reset(new TraceSetting(
-      level, rate, count, log_frequency, file, false, false, false, false,
-      false));
+      level, rate, count, log_frequency, file, mode, config_map,
+      false /*level_specified*/, false /*rate_specified*/,
+      false /*count_specified*/, false /*log_frequency_specified*/,
+      false /*filepath_specified*/, false /*mode_specified*/,
+      false /*config_map_specified*/));
   global_setting_.reset(new TraceSetting(
-      level, rate, count, log_frequency, file, false, false, false, false,
-      false));
+      level, rate, count, log_frequency, file, mode, config_map,
+      false /*level_specified*/, false /*rate_specified*/,
+      false /*count_specified*/, false /*log_frequency_specified*/,
+      false /*filepath_specified*/, false /*mode_specified*/,
+      false /*config_map_specified*/));
```

Comment:
Ideally this will be encapsulated in a config struct or similar in a follow-up PR

Reply:
Yes, I've discussed that with @tanmayv25  and created a ticket for this